source/common/ssl/context_config_impl.cc

```diff
@@ -99,13 +136,66 @@ ContextConfigImpl::ContextConfigImpl(
           RepeatedPtrUtil::join(config.tls_params().cipher_suites(), ":"), DEFAULT_CIPHER_SUITES)),
       ecdh_curves_(StringUtil::nonEmptyStringOrDefault(
           RepeatedPtrUtil::join(config.tls_params().ecdh_curves(), ":"), DEFAULT_ECDH_CURVES)),
-      tls_certficate_provider_(getTlsCertificateConfigProvider(config, factory_context)),
-      certficate_validation_context_provider_(
-          getCertificateValidationContextConfigProvider(config, factory_context)),
+      tls_certficate_provider_(
+          getTlsCertificateConfigProvider(config, factory_context, &tls_certificate_config_)),
+      certficate_validation_context_provider_(getCertificateValidationContextConfigProvider(
+          config, factory_context, &validation_context_config_, &default_cvc_)),
       min_protocol_version_(
           tlsVersionFromProto(config.tls_params().tls_minimum_protocol_version(), TLS1_VERSION)),
-      max_protocol_version_(tlsVersionFromProto(config.tls_params().tls_maximum_protocol_version(),
-                                                TLS1_2_VERSION)) {}
+      max_protocol_version_(
+          tlsVersionFromProto(config.tls_params().tls_maximum_protocol_version(), TLS1_2_VERSION)) {
+  if (default_cvc_) {
+    cvc_validation_callback_handle_ =
+        dynamic_cast<Secret::CertificateValidationContextSdsApi*>(
+            certficate_validation_context_provider_.get())
+            ->addValidationCallback(
+                [this](const envoy::api::v2::auth::CertificateValidationContext& dynamic_cvc) {
+                  getCombinedValidationContextConfig(dynamic_cvc);
+                  validation_context_config_.reset();
```

Comment:
dont free validation_context_config_ here.  In case this config is bad (bad from other default config impl, not from this one),  we still want to use old config.

Reply:
Good point. Removed. Thanks.